agent_app_sample_code/validators/validate_agent_config.py

```diff
@@ -0,0 +1,81 @@
+# Databricks notebook source
+# Install these if running this Notebook on its own
+# %pip install -U -qqqq databricks-sdk mlflow-skinny
+# dbutils.library.restartPython()
+
+# COMMAND ----------
+
+from databricks.sdk import WorkspaceClient
+from databricks.sdk.service.vectorsearch import EndpointStatusState, EndpointType
+from mlflow.utils import databricks_utils as du
+from databricks.sdk.service.serving import EndpointCoreConfigInput, EndpointStateReady
+from databricks.sdk.errors import ResourceDoesNotExist
+import os
+from databricks.sdk.service.compute import DataSecurityMode
+from pyspark.sql import SparkSession
+
+w = WorkspaceClient()
+browser_url = du.get_browser_hostname()
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ## Check if Vector Search Index exists
+
+# COMMAND ----------
+
+# Create the Vector Search endpoint if it does not exist
+from databricks.sdk import WorkspaceClient
+from databricks.sdk.service.vectorsearch import EndpointType
+from databricks.sdk.errors import ResourceDoesNotExist
+
+w = WorkspaceClient()
+
+
+def validate_retriever_config(retriever_config: AgentConfig):
+    try:
+        w.vector_search_indexes.get_index(retriever_config.vector_search_index)
+        print(f"PASS: {retriever_config.vector_search_index} exists.")
+    except ResourceDoesNotExist as e:
+        print(
+            f"FAIL: {retriever_config.vector_search_index} does not exist.  Please check the vector search index name."
+        )
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ### Check LLM endpoint
+
+# COMMAND ----------
+
+# TODO: Add check for tool calling support
```

Comment:
? QQ I don't fully follow what this one means, any chat LLM technically supports tool use right?

Reply:
My understanding is that support for tool calling is limited to a handful of models on Databricks at the moment